src/ccutil/tessdatamanager.cpp

```diff
@@ -162,12 +188,16 @@ void TessdataManager::OverwriteEntry(TessdataType type, const char *data, int si
 
 // Saves to the given filename.
 bool TessdataManager::SaveFile(const char *filename, FileWriter writer) const {
-  // TODO: This method supports only the proprietary file format.
+// TODO: This method supports only the proprietary file format.
   ASSERT_HOST(is_loaded_);
   std::vector<char> data;
   Serialize(&data);
   if (writer == nullptr) {
+#if defined(HAVE_LIBARCHIVE)
+    return SaveArchiveFile(filename);
```

Comment:
With this change `TessdataManager::SaveFile` will always write traineddata files in ZIP format which are incompatible with Tesseract binaries which were build without LibArchive. I'm afraid that would cause problems for a lot of people.

Reply:
I thought libarchive can deduce archive types.

Reply:
Yes, it can, but the current code uses `archive_write_set_format_zip`. And of course libarchive cannot write the proprietary traineddata format.